### PR TITLE
[AppConfig] Refactor tests

### PR DESCRIPTION
**Pull request by YalinLi0312**
First part updates for issue #https://github.com/Azure/azure-sdk-for-python/issues/21146

### CODE REVIEW

File: sdk/appconfiguration/azure-appconfiguration/tests/test_azure_configuration_client.py [hunk 1]
```diff
@@ -212,132 +177,135 @@ def test_get_configuration_setting_label(self, client, appconfiguration_connecti
...
+        client.delete_configuration_setting(key=compare_kv.key, label=compare_kv.label)
 
     # method: delete_configuration_setting
     @app_config_decorator
-    def test_delete_with_key_no_label(self, client, appconfiguration_connection_string, test_config_setting_no_label):
-        to_delete_kv = test_config_setting_no_label
-        client.delete_configuration_setting(to_delete_kv.key)
+    def test_delete_with_key_no_label(self, appconfiguration_connection_string):
+        client = self.create_client(appconfiguration_connection_string)
+        to_delete_kv = self.create_config_setting_no_label()
```
Comment (xiangyan99) [inline sdk/appconfiguration/azure-appconfiguration/tests/test_azure_configuration_client.py:198]: Create or delete?

File: sdk/appconfiguration/azure-appconfiguration/tests/test_azure_configuration_client.py [hunk 2]
```diff
@@ -212,132 +177,135 @@ def test_get_configuration_setting_label(self, client, appconfiguration_connecti
...
 
     @app_config_decorator
-    def test_delete_with_key_label(self, client, appconfiguration_connection_string, test_config_setting, test_config_setting_no_label):
-        to_delete_kv = test_config_setting
-        client.delete_configuration_setting(
-            to_delete_kv.key, label=to_delete_kv.label
-        )
+    def test_delete_with_key_label(self, appconfiguration_connection_string):
+        client = self.create_client(appconfiguration_connection_string)
+        to_delete_kv = self.create_config_setting()
```
Comment (xiangyan99) [inline sdk/appconfiguration/azure-appconfiguration/tests/test_azure_configuration_client.py:205]: Same

File: sdk/appconfiguration/azure-appconfiguration/tests/test_azure_configuration_client.py (hunk 1 above)
Comment (YalinLi0312) [reply sdk/appconfiguration/azure-appconfiguration/tests/test_azure_configuration_client.py:198]: create! I removed the `create` and `add two test config settings` in wrapper.

File: sdk/appconfiguration/azure-appconfiguration/tests/test_azure_configuration_client.py (hunk 2 above)
Comment (YalinLi0312) [reply sdk/appconfiguration/azure-appconfiguration/tests/test_azure_configuration_client.py:205]: It is `create` here as well, just keep the same variable name. I'm free to rename if you think it's confusing.

Comment (xiangyan99) [reply sdk/appconfiguration/azure-appconfiguration/tests/test_azure_configuration_client.py:205]: But the test name is test_delete_with_key_label. Did I miss something?

Comment (YalinLi0312) [reply sdk/appconfiguration/azure-appconfiguration/tests/test_azure_configuration_client.py:205]: Oh, you are right, I'll add a config_setting then test `delete`

**Review by xiangyan99 — APPROVED** [review]

**Review by mccoyp — APPROVED** [review]

**Pull request merged**